Expand WCPay Switch incentives to more countries

ismaeldcom opened the pull request:
### Draft TODO
- [x] New designs. (EJyQdHyExlXuiniHADVXEC-fi-39_7921)
- [ ] New eligibility for merchants (pbUcTB-2yV-p2):
  - Update supported countries.
  - Has orders
  - Remove gateway check, not having WCPay is enough.
- [ ] New promo codes from server.
- [ ] Dynamic incentive copy showing the right incentive %.
- [ ] The incentive page FAQ questions copy is updated to reflect the new conditions.


### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- If necessary, indicate if this PR is part of a bigger feature. Add a label with the format `focus: name of the feature [team:name of the team]`. -->

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

Closes https://github.com/Automattic/woocommerce-payments-server/issues/3479

<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Include detailed instructions on how these changes can be tested. Review and follow the guide for how to write high-quality testing instructions. -->

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1.
2.
3.

<!-- End testing instructions -->

ismaeldcom committed:
Initial refactor of components

Move out from index every UI component to make it easier to maintain.

ismaeldcom committed:
Initial implementation of new design

Some minor adjustments are missing, and I've added a few TODOs to complete later.

ismaeldcom committed:
Add changelog entry

ismaeldcom committed:
Finish design update

ismaeldcom committed:
Compress menu icon

ismaeldcom committed:
Add user's first name to heading copy

vladolaru commented:
Excellent work @ismaeldcom. I've updated [the issue](https://github.com/Automattic/woocommerce-payments-server/issues/3479) with more up-to-date acceptance criteria.

ismaeldcom committed:
Remove other gateway condition

Not having WCPay is the new eligibility condition for this case.

ismaeldcom committed:
Update country eligibility to use and array instead of a single one.

The list of countries will change and maybe even be handled differently from server side data.

ismaeldcom committed:
Add has processed orders condition

By checking the number of orders with `wc-completed` status

ismaeldcom committed:
Fetch and cache eligible incentive from WCPay API

Sharing its properties under `window.WcSettings.admin`

ismaeldcom committed:
Filter `allowed_promo_notes`

To be able to add those form WCPay API request

ismaeldcom committed:
Update JS side to use the new shared incentive details
